src/tox.rs

```diff
@@ -529,6 +529,15 @@ fn tox_loop(
 
     let mut tox = Tox::new(tox_options, None).unwrap();
 
+    {
+        let sk = tox.get_secret_key();
+        let ev = Event::SecretKey {
+            secret_key: format!("{}", sk)
+        };
+
+        drop(answer_tx.try_send(Answer::Event(ev)))
```

Comment:
why try_send?

Reply:
Also I thought you'd send it in main right after `let (sink, stream) = s.split();`

Reply:
Because this is a tokio channel: https://docs.rs/tokio/0.1.20/tokio/sync/mpsc/struct.Sender.html#method.try_send

Reply:
`Tox` generates the secret key.

Reply:
OK